cmd/report_writer.go

```diff
@@ -0,0 +1,29 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+func reportWriter() (io.Writer, func() error, error) {
+	nop := func() error { return nil }
+
+	path := strings.TrimSpace(appConfig.File)
+	switch len(path) {
+	case 0:
+		return os.Stdout, nop, nil
+	default:
+		reportFile, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+		if err != nil {
+			return nil, nop, fmt.Errorf("unable to create report file: %w", err)
+		}
+		return reportFile, func() error {
+			if !appConfig.Quiet {
+				fmt.Printf("Report written to %q\n", path)
+			}
+			return reportFile.Close()
+		}, nil
+	}
```

Comment:
very small nit: This is a good example of code that's well structured, but could be more readable if vertical spacing were leveraged more. Blank lines to separate subsections of thoughts can be really nice for readers looking to get a sense of structure, particular with regard to control flow — e.g., using blank lines to make `return` lines more clear.

Reply:
https://github.com/anchore/syft/pull/536 and https://github.com/anchore/grype/pull/443